### PR TITLE
[MFT] Propagation of MCLabels, use of SMatrix on TrackMFT

### PR DESCRIPTION
**Pull request by rpezzi**
* Added propagation of MCLabels to fitted MFT Tracks and print function
* FitterWorkflow now uses gsl::span
* TrackMFT dropped TMatrix in favor of SMatrix

### CODE REVIEW

File: Detectors/ITSMFT/MFT/workflow/src/TrackFitterSpec.cxx [hunk 1]
```diff
@@ -163,13 +170,32 @@ void convertTrack(const T& inTrack, O& outTrack, C& clusters)
...
+  outTrack.setMCCompLabels(inTrack.getMCCompLabels(), nClusters);
   //std::cout << "  ** outTrack has getNClusters = " << outTrack.getNClusters() << std::endl;
   //for (auto par = outTrack.rbegin(); par != outTrack.rend(); par++)
   //  std::cout << "     getZ() =  " << par->getZ() << std::endl;
 
   //fit(outTrack, false);
 }
 
+//_________________________________________________________________________________________________
+SMatrix55 TtoSMatrix55(TMatrixD inMatrix)
```
Comment (shahor02) [inline Detectors/ITSMFT/MFT/workflow/src/TrackFitterSpec.cxx:182]: @rpezzi will be more efficient to use sym. matrix 
````
using SMatrix55 = ROOT::Math::SMatrix<double, 5,53, ROOT::Math::MatRepSym<double, 5>>;
...
// TMatrix to sym SMatrix
for (auto i = 5; i--;) {
    outMatrix(i, i) = inMatrix(i, i);
    for (auto j = i; j--;) {
        outMatrix(i, j) = inMatrix(i, j);
    }
}

// sym SMatrix to TMatrix
for (auto i = 5; i--;) {
   outMatrix(i, i) = inMatrix(i, i);
    for (auto j = i; j--;) {
        outMatrix(j, i) = outMatrix(i, j) = inMatrix(i, j);
    }
}

````

Comment (rpezzi) [reply Detectors/ITSMFT/MFT/workflow/src/TrackFitterSpec.cxx]: Thanks. Done.

**Comment by rpezzi** [comment]
Hi @shahor02 , thanks for your valuable input. For now I was just copying the methods from TrackCA and TrackLTF, which will eventually be discarded from storage. This way, the macros I use to assess the tracking performance do not break. I fully agree with the need to change the way the MFT deal with MC Labels (and clusters). I suggest we keep the MC Labels to a future PR.

**Comment by shahor02** [comment]
Thanks @rpezzi ! OK, so, can this be merged now?

**Pull request merged**